### PR TITLE
feat!: support PostgreSQL as optional storage backend

### PR DESCRIPTION
**Pull request by jsstevenson**
Finalizes a dev release implementing an optional Postgres DB backend to use in TheraPy CI. Once tests are passing for everything, there will be another full release.

Otherwise, everything is basically the same as the gene PR. I've noticed that all three normalizers have slightly different query method structure, which would be good to fix up at some point, but not pressing rn.

### CODE REVIEW

**Review by korikuzma — CHANGES_REQUESTED** [review]
Overall looks great!

**Comment by korikuzma** [comment]
@jsstevenson It also looks like there's some conflicts that need resolved before merging

File: disease/cli.py [hunk 1]
```diff
@@ -21,222 +20,269 @@
...
+    command line option or DISEASE_NORM_REMOTE_DB_URL environment variable.
+
+    \f
+    :param data_url: user-specified location to pull DB dump from
+    :param db_url: URL to normalizer database
+    """  # noqa: D301
+    if not click.confirm("Are you sure you want to overwrite existing data?"):
+        click.get_current_context().exit()
+    if not data_url:
+        data_url = os.environ.get("DISEASE_NORM_REMOTE_DB_URL")
```
[the remark this answers is not in the document]
Comment (jsstevenson) [reply disease/cli.py:64]: Yeah, I want to do one more iteration on the gene docs and then I'll start building out comparable stuff with disease and therapy

File: .github/workflows/github-actions.yml [hunk 2]
```diff
@@ -8,32 +8,51 @@ jobs:
...
       with:
-          python-version: ${{ matrix.python-version }}
-
-    - name: Install dependencies
-      run: python3 -m pip install ".[dev,test]"
+        python-version: ${{ matrix.python-version }}
 
-    - name: Build local DynamoDB server
+    - name: Build local DynamoDB
+      if: ${{ env.GENE_NORM_DB_URL == 'http://localhost:8000' }}
```
Comment (korikuzma) [inline .github/workflows/github-actions.yml:35]:
```suggestion
      if: ${{ env.DISEASE_NORM_DB_URL == 'http://localhost:8000' }}
```

File: .github/workflows/github-actions.yml [hunk 3]
```diff
@@ -8,32 +8,51 @@ jobs:
...
       run: |
-          chmod +x ./tests/scripts/dynamodb_run.sh
-          ./tests/scripts/dynamodb_run.sh
+        chmod +x ./tests/unit/dynamodb_build.bash
+        ./tests/unit/dynamodb_build.bash
+
+    - name: Install DynamoDB dependencies
+      if: ${{ env.GENE_NORM_DB_URL == 'http://localhost:8000' }}
```
Comment (korikuzma) [inline .github/workflows/github-actions.yml:41]:
```suggestion
      if: ${{ env.DISEASE_NORM_DB_URL == 'http://localhost:8000' }}
```

File: .github/workflows/github-actions.yml [hunk 4]
```diff
@@ -8,32 +8,51 @@ jobs:
...
+      run: python3 -m pip install ".[etl,test]"
+
+    - name: Install PG dependencies
+      if: ${{ env.GENE_NORM_DB_URL != 'http://localhost:8000' }}
```
Comment (korikuzma) [inline .github/workflows/github-actions.yml:45]:
```suggestion
      if: ${{ env.DISEASE_NORM_DB_URL != 'http://localhost:8000' }}
```

**Review by korikuzma — CHANGES_REQUESTED** [review]

File: .github/workflows/github-actions.yml (hunk 4 above)
Comment (jsstevenson) [reply .github/workflows/github-actions.yml:45]: f

**Review by korikuzma — APPROVED** [review]

**Review by korikuzma — APPROVED** [review]

**Pull request merged**